src/bp/ui-studio/src/web/views/FlowBuilder/nodeProps/ConditionModalForm.jsx

```diff
@@ -29,17 +42,62 @@ export default class ConditionModalForm extends Component {
       this.setState({
         typeOfTransition,
         condition: item.condition,
+        conditionType: this.getConditionType(item.condition),
         flowToSubflow: typeOfTransition === 'subflow' ? item.node : null,
         flowToNode: typeOfTransition === 'node' ? item.node : null,
         returnToNode: typeOfTransition === 'return' ? item.node.substr(1) : ''
       })
     } else {
-      this.resetForm({ condition: (item && item.condition) || '' })
+      const condition = (item && item.condition) || ''
+      this.resetForm({
+        condition,
+        conditionType: this.getConditionType(condition)
+      })
     }
 
     this.setState({ isEdit: Boolean(item) })
   }
 
+  getConditionType(condition) {
+    condition = condition.trim()
+
+    if (condition === 'true') {
+      return 'always'
+    } else if (condition.includes('event.nlu.intent.name')) {
+      this.extractIntent(condition)
+      return 'intent'
+    } else if (availableProps.some(props => condition.indexOf(props.value) === 0)) {
+      this.extractProps(condition)
```

Comment:
This method shouldn’t have side effects if it’s named “get”

Reply:
good point